quickwit/quickwit-proto/src/ingest/ingester.rs

```diff
@@ -17,12 +17,20 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-use crate::types::{queue_id, Position, QueueId};
+use crate::types::{queue_id, Position, QueueId, ShardId};
 
 include!("../codegen/quickwit/quickwit.ingest.ingester.rs");
 
 pub use ingester_service_grpc_server::IngesterServiceGrpcServer;
 
+impl FetchEof {
+    pub fn shard_id(&self) -> ShardId {
```

Comment:
if not copy we should probably change it to 
```suggestion
    pub fn shard_id(&self) -> &ShardId {
```

Reply:
Good point.